plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -863,6 +863,56 @@ public List<String> getAllViews(String databaseName)
         }
     }
 
+    @Override
+    public List<SchemaTableName> getAllTables()
+    {
+        try {
+            return retry()
+                    .stopOn(UnknownDBException.class)
+                    .stopOnIllegalExceptions()
+                    .run("getAllTables", stats.getGetAllTables().wrap(() -> {
+                        try (ThriftMetastoreClient client = createMetastoreClient()) {
+                            return client.getAllTables();
+                        }
+                    }));
+        }
+        catch (TException e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public List<SchemaTableName> getAllViews()
+    {
+        try {
+            return retry()
+                    .stopOn(UnknownDBException.class)
+                    .stopOnIllegalExceptions()
+                    .run("getAllViews", stats.getGetAllViews().wrap(() -> {
+                        try (ThriftMetastoreClient client = createMetastoreClient()) {
+                            if (translateHiveViews) {
+                                return client.getAllViews();
+                            }
+                            // Without translateHiveViews, the views are actually Trino flagged tables,
+                            // and they will not be properly returned by the client's getAllViews method
+                            return getAllDatabases().stream()
+                                    .flatMap(db -> getAllTables(db).stream()
```

Comment:
Shouldn't this be `getTablesWithParameter` and we pass `PRESTO_VIEW` flag.

Reply:
It's done instead in `getAllViews` method, which should be invoked here instead of `getAllTables`. I will push the fix soon.